### ISSUE TITLE
Unstable indices download

### ISSUE THREAD

**Issue by coolkal** [hemenkapadia/getbhavcopy#86]
Although a lot of NSE indices have been added in Ver 3, the downloading of these indices seem to be quite unstable. Almost with every download some indices are skipped There also seems to be an element of randomness on which ones are skipped Like when i download data 3-4 times for a particular day clicking the download button, more often than not different indices are skipped with each pass or new indices are added to the original one/s skipped.

Doesn't matter so much when the less important indices are skipped but sometimes important ones are skipped too Like the NSENIFTY index for 11/04/2019 The error shown in all such skips 'NSE server did not return data for the index' but when i check NSE site the data is very much there (in indices section) In fact i manually downloaded & updated Nifty data for 11/04/2019.

If so many indices are difficult to handle please give users a choice on which indices should be downloaded Very few if any users will need data for all of the myriad indices.

**Comment by hemenkapadia** [comment]
Hi @coolkal, thanks for your feedback. This is not an issue in Getbhavcopy, but rather an issue with NSE server whereby it randomly fails to return a response. I can enable some retry mechanism, but then there is no guarantee that it will succeed too.

Will accept this feedback and see what can be done about this.

**Comment by coolkal** [comment]
Last couple of weeks indices download was perfect I thought the retry mechanism was already built in (on server side) & working Today though it's again bit of a mess Arbitrary skipping is back After 3 attempts managed to get all sectoral indices

**Comment by hemenkapadia** [comment]
No, the retry version will not get deployed unless I implement it in a version. I have not implemented it so far. What you are experiencing is directly related to NSE server instability. Actually, if I think over it, implementing the retry mechanism may not be super helpful is the server itself is having issues.

**Comment by hemenkapadia** [comment]
Providing an option to download selective indices is a good suggestion which I can consider.
